### PR TITLE
MQE-604: support new  package types (magento2-test-module and magento…

### PR DESCRIPTION
**Pull request by jilu1**
…2-test-component).

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento-composer-installer?pullRequest=19) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento-composer-installer?pullRequest=19) before we can accept your contribution.<br/>

**Comment by buskamuza** [comment]
1. Why do we need both `magento2-test-module` and `magento2-test-component` if they're installed in the same location? What's the difference?
2. The names are too generic. They potentially block support of other tests (e.g., integration). Please consider using "functional" in the name

File: src/MagentoHackathon/Composer/Magento/Plugin.php [hunk 1]
```diff
@@ -311,7 +311,12 @@ private function saveVendorDirPath(Composer $composer)
             realpath($composer->getConfig()->get('vendor-dir')),
             true
         );
-        $vendorPathFile = $magentoDir . '/app/etc/vendor_path.php';
+        $etcDir = $magentoDir . '/app/etc';
+        if (file_exists($etcDir)) {
+            $vendorPathFile = $etcDir . '/vendor_path.php';
+        } else {
+            $vendorPathFile = $magentoDir . '/vendor_path.php';
```
Comment (buskamuza) [inline src/MagentoHackathon/Composer/Magento/Plugin.php]: Why is this change needed? What use case does it cover?

Comment (jilu1) [reply src/MagentoHackathon/Composer/Magento/Plugin.php]: This is needed when composer installer is used for functional acceptance test modules, where `app/code` is not available.

**Comment by jilu1** [comment]
I am aware of this and the reason for two names is to be consistence with magento2-module and magento2-component. Although in that case, they are installed to different locations. Let me know your thoughts on this?
I can add `functional` in the name.

**Comment by paliarush** [comment]
@jilu1 Could you please explain why do we need custom marshaling here?

**Comment by jilu1** [comment]
@paliarush Because we decided that MFTF will not support test modules in vendor directory.

**Comment by vkublytskyi** [comment]
This PR is not active for 1 year, proposed changes not explained well and motivation seems outdated as now we strongly recommend to put all tests together with modules. Existing of monolith folder dev/tests contradicts modularity goal of Magento and should be eliminated eventually.

If this PR is still needed for some reason please reopen it and provide more explanation.
